ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryIdConverterFactory.java

```diff
@@ -181,7 +183,8 @@ private CompletionStage<String> convertId(String input, RestRequest restRequest,
           // Set named blob state as 'IN_PROGRESS', will set the state to be 'READY' in the ttlUpdate success callback: routerTtlUpdateCallback
           state = NamedBlobState.IN_PROGRESS;
         }
-        conversionFuture = getNamedBlobDb().put(record, state, RestUtils.isUpsertForNamedBlob(restRequest.getArgs())).thenApply(result -> result.getInsertedRecord().getBlobId());
+        conversionFuture = getNamedBlobDb().put(record, state, RestUtils.isUpsertForNamedBlob(restRequest.getArgs())).thenApply(
+            result -> result.getInsertedRecord().getBlobId() + NAMED_BLOBID_VERSION_SEPARATOR + result.getInsertedRecord().getVersion());
```

Comment:
Let's not change the return value of id converter, let's put the version in the RestRequest. We can add an InternalKey in the RestUtils and save the version as the value of this internal key, so that the NamedBlobPutHandler can get the version out.

Reply:
Good idea, I just updated with that.